src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -1,22 +1,79 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Buffers;
+using System.Diagnostics;
 using System.IO;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System
 {
     internal sealed class NSLogStream : ConsoleStream
     {
-        public NSLogStream() : base(FileAccess.Write) {}
+        private readonly StringBuilder _buffer = new StringBuilder();
+        private readonly Decoder _decoder;
+
+        public NSLogStream() : base(FileAccess.Write)
+        {
+            _decoder = Encoding.Unicode.GetDecoder();
+        }
 
         public override int Read(Span<byte> buffer) => throw Error.GetReadNotSupported();
 
         public override unsafe void Write(ReadOnlySpan<byte> buffer)
         {
-            fixed (byte* ptr = buffer)
+            int charCount = _decoder.GetCharCount(buffer, false);
+            char[] pooledBuffer = ArrayPool<char>.Shared.Rent(charCount);
+            Span<char> charSpan = pooledBuffer;
```

Comment:
One "gotcha" with ArrayPool is that the length of the rented array is guaranteed to be _at least_ the requested length, but it may (and likely will) be larger than what you requested. So it is not a good idea to use the length of the rented array as the same as the requested length.

Below in `WriteOrCache`, it is using `charBuffer.Length`, which will be the length of the rented array.

Reply:
thanks, I used the length given by the decoder